src/nccl_ofi_net.c

```diff
@@ -950,8 +948,17 @@ static ncclResult_t ofi_connect(int dev, void *handle, void **sendComm)
 
 	/* Send "connect" message to remote EP */
 	do {
+		nccl_ofi_req_t *req = allocate_nccl_ofi_request(sComm->nccl_ofi_reqs_fl);
+
+		if (OFI_UNLIKELY(req == NULL)) {
+			ret = ncclSystemError;
+			NCCL_OFI_WARN("Unable to get NCCL OFI request for device %d",
+				     sComm->dev);
+			goto error;
+		}
```

Comment:
This patch currently breaks the unit tests for other providers like 'tcp;ofi_rxm'. The error is a segmentation fault introduced while freeing this allocated NCCL request.

After allocating the request, we need to do the following:
```
+               req->direction = NCCL_OFI_SEND;
+               req->sComm = sComm;
```

Reply:
Oh, thanks. Fixed this bug